presto-main/src/main/java/com/facebook/presto/operator/window/PagesWindowIndex.java

```diff
@@ -80,6 +82,14 @@ public Slice getSlice(int channel, int position)
     }
 
     @Override
+    public Block getBlock(int channel, int position)
+    {
+        BlockBuilder builder = pagesIndex.getType(channel).createBlockBuilder(new BlockBuilderStatus(), 1);
```

Comment:
Rewritten (5ad5fd65979cda713c7da3789d2e78003864a5f9) to use `Block.getSingleValueBlock` instead of `Type.appendTo`. Though, both approaches are equal in terms of copying memory.